polling_stations/apps/addressbase/models.py

```diff
@@ -19,14 +19,13 @@ class AbstractAddress(models.Model):
     address = models.TextField(blank=True)
     postcode = models.CharField(blank=True, max_length=15, db_index=True)
     location = models.PointField(null=True, blank=True)
-    objects = AddressManager()
 
     class Meta:
         abstract = True
 
 
 class Address(AbstractAddress):
-    pass
+    objects = AddressManager()
```

Comment:
Can you explain the reason for this? Does this mean that all sub-classes of `AbstractAddress` will need to do this?

Reply:
The reason for moving the manager declaration on to the implementation in that commit is because the methods `postcodes_for_district()` and `points_for_postcode()` are specific to WhereDoIVote (they're also marked for removal in future work - the `AddressWithPollingStation` model won't use them) so I didn't want to transfer them to the `uk_geo_utils` app. You would need to add that declaration to any implementation of `AbstractAddress` which should inherit those methods, but `AbstractAddress` itself shouldn't have them.

In a [later commit](https://github.com/DemocracyClub/UK-Polling-Stations/pull/1063/commits/98603881f176ae426d9589bfd82f5e9df620cf31) I [added a `uk_geo_utils.models.AbstractAddressManager` class](https://github.com/DemocracyClub/UK-Polling-Stations/pull/1063/commits/98603881f176ae426d9589bfd82f5e9df620cf31#diff-9ab660099325f43819c4b7cc51ee6780) with a method that all `Address` models should have and [modified `addressbase.models.AddressManager` to extend it](https://github.com/DemocracyClub/UK-Polling-Stations/pull/1063/commits/98603881f176ae426d9589bfd82f5e9df620cf31#diff-2a693a38b300d697ef456543e04467c2)

Reply:
👍